x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go

```diff
@@ -78,7 +78,7 @@ func (k msgServer) VoteOnObservedOutboundTx(goCtx context.Context, msg *types.Ms
 		// Set this the first time when the ballot is created
 		// The ballot might change if there are more votes in a different outbound ballot for this cctx hash
 		cctx.GetCurrentOutTxParam().OutboundTxBallotIndex = ballotIndex
-		k.SetCrossChainTx(ctx, cctx)
+		//k.SetCctxAndNonceToCctxAndInTxHashToCctx(ctx, cctx)
```

Comment:
It might be better to split this function  `SetCctxAndNonceToCctxAndInTxHashToCctx ` into three. 
We can always set the CCTX here and keep the other two mappings the same.

SetCctxAndNonceToCctxAndInTxHashToCctx can be a helper function which calls the three, but not use it here

Reply:
I don't think splitting SetCctxAndNonceToCctxAndInTxHashToCctx() into three is good idea:
1. the three "sub-functions" are always grouped and called together in 6 call sites, except this one; 
2. The function is already very short; 
3. If you want to set CCTX just set CCTX. There is no need for creating additional functions.

Reply:
We don't have a set SetCCTX function right now, 
`SetCctxAndNonceToCctxAndInTxHashToCctx` can exist and call the three sub-functions.
The function is short but still has logic, which is different from what a setter function should do. 

Having three subfunctions will also make it easier to write unit tests.

But I understand that we are short on time and this needs to be merged. Approving for now as functionality wise the changes are minor